### PR TITLE
Gitlab: separate gitlab client and provider

### PR DESCRIPTION
**Pull request by domizei385**
## Problem Statement

What is the problem you're trying to solve?

## Related Issue

Fixes https://github.com/external-secrets/external-secrets/issues/2185

## Proposed Changes

How do you like to solve the issue and why?

## Checklist

- [ ] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [ ] All commits are signed with `git commit --signoff`
- [ ] My changes have reasonable test coverage
- [ ] All tests pass with `make test`
- [ ] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by domizei385** [comment]
@moolen Hi. Care to have a look whether this will be enough for #2185 ?

File: pkg/provider/gitlab/gitlab.go [hunk 1]
```diff
@@ -69,25 +68,14 @@ type GroupVariablesClient interface {
 	ListVariables(gid interface{}, opt *gitlab.ListGroupVariablesOptions, options ...gitlab.RequestOptionFunc) ([]*gitlab.GroupVariable, *gitlab.Response, error)
 }
 
-// Gitlab Provider struct with reference to GitLab clients, a projectID and groupIDs.
+// Gitlab Provider struct with gitlab auth, environment, projectID and groupIDs.
 type Gitlab struct {
```
Comment (moolen) [inline pkg/provider/gitlab/gitlab.go:72]: This is unused, can be removed

File: pkg/provider/gitlab/gitlab.go [hunk 2]
```diff
@@ -118,104 +100,51 @@ func (c *gClient) setAuth(ctx context.Context) error {
...
+	credentials := credentialsSecret.Data[c.store.Auth.SecretRef.AccessToken.Key]
+	if credentials == nil || len(credentials) == 0 {
+		return nil, fmt.Errorf(errMissingSAK)
 	}
-	return nil
+	return credentials, nil
 }
 
 // Function newGitlabProvider returns a reference to a new instance of a 'Gitlab' struct.
 func NewGitlabProvider() *Gitlab {
```
Comment (moolen) [inline pkg/provider/gitlab/gitlab.go:121]: This can also be removed.

**Review by moolen — COMMENTED** [review]
I did a quick test, this indeed solves the issue :tada: . Thank you so much for picking it up swiftly and proposing a solution :heart: 

The `Gitlab` type can be removed, see comments below. I know this is a draft, i thought  i'd leave this here anyways.

**Comment by domizei385** [comment]
You comments and effort are very welcome. I quickly drafted this and didn't really get around testing this.

**Comment by domizei385** [comment]
@moolen No idea, what the pipeline is complaining about. make fmt does not trigger any result.

**Comment by moolen** [comment]
I can take a look at it next week. Could be a go/linter version issue

**Review by moolen — APPROVED** [review]
Did some manual tests against gitlab.com, LGTM :+1:  

Thank you for providing a fix, much appreciated :bow:

**Pull request merged**